src/main/java/core/basesyntax/strategy/mapvalidator/KeyPresentValidator.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.mapvalidator;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public interface KeyPresentValidator extends MapValidator {
+    @Override
+    default void validateMap(Map<String, Integer> fruitQuantityMap,
+                                    FruitTransaction transaction) {
+        if (!fruitQuantityMap.containsKey(transaction.getFruit())) {
```

Comment:
The `validateMap` method implementation should not be in the interface. Interfaces are meant to define contracts, not to provide implementation. Consider moving this default method implementation to an abstract class or a concrete class that implements this interface.